### PR TITLE
Minor lint tweaks

### PR DESCRIPTION
**Pull request by XhmikosR**

### CODE REVIEW

File: remark-lint-nodejs-yaml-comments.js [hunk 1]
```diff
@@ -86,9 +90,13 @@ function invalidChangesKeys(change) {
     if (keys[index] !== changesExpectedKeys[index]) return true;
   }
 }
+
 function validateSecurityChange(file, node, change, index) {
   if ("commit" in change) {
-    if (typeof change.commit !== "string" || isNaN(`0x${change.commit}`)) {
+    if (
+      typeof change.commit !== "string" ||
+      Number.isNaN(`0x${change.commit}`)
```
Comment (XhmikosR) [inline remark-lint-nodejs-yaml-comments.js:98]: Switched this to `Number.isNan`

File: remark-lint-nodejs-links.js [hunk 2]
```diff
@@ -1,13 +1,14 @@
-import fs from "fs";
-import path from "path";
-import { pathToFileURL } from "url";
+import fs from "node:fs";
```
Comment (XhmikosR) [inline remark-lint-nodejs-links.js:1]: I can revert these changes if you prefer

File: test/test.js [hunk 3]
```diff
@@ -1,13 +1,12 @@
-import assert from "assert";
-import fs from "fs";
-import process from "process";
-
+import assert from "node:assert/strict";
```
Comment (XhmikosR) [inline test/test.js:1]: Switched to strict mode

File: index.js [hunk 4]
```diff
@@ -25,14 +25,14 @@ import remarkLintNoShellDollars from "remark-lint-no-shell-dollars";
...
-import remarkLintNodejsLinks from "./remark-lint-nodejs-links.js";
-import remarkLintNodejsYamlComments from "./remark-lint-nodejs-yaml-comments.js";
 import remarkLintProhibitedStrings from "remark-lint-prohibited-strings";
 import remarkLintRuleStyle from "remark-lint-rule-style";
 import remarkLintStrongMarker from "remark-lint-strong-marker";
 import remarkLintTableCellPadding from "remark-lint-table-cell-padding";
 import remarkLintTablePipes from "remark-lint-table-pipes";
 import remarkLintUnorderedListMarkerStyle from "remark-lint-unordered-list-marker-style";
+import remarkLintNodejsYamlComments from "./remark-lint-nodejs-yaml-comments.js";
+import remarkLintNodejsLinks from "./remark-lint-nodejs-links.js";
```
Comment (Trott) [inline index.js:35]: I understand the logic of moving these to the end, but previously the entire list had been in alphabetical order, which makes it very easy to determine if something has or has not been added with visual inspection. Now we effectively have two lists. I guess I'm -0 on this, but if we are going to do it, perhaps we can add a blank line between the lists to make it visually clear where one alphabetical list/section ends and the other one begins?

Comment (XhmikosR) [reply index.js]: I reverted this since it's not enforce anyway :)

File: remark-lint-nodejs-yaml-comments.js [hunk 5]
```diff
@@ -110,14 +119,13 @@ function validateSecurityChange(file, node, change, index) {
...
+
 function validateChanges(file, node, changes) {
   if (!Array.isArray(changes))
     return file.message("`changes` must be a YAML list", node);
 
   const changesVersions = [];
-  for (let index = 0; index < changes.length; index++) {
-    const change = changes[index];
-
+  for (const [index, change] of changes.entries()) {
```
Comment (aduh95) [inline remark-lint-nodejs-yaml-comments.js:128]: What's the lint rule behind this change?

Comment (XhmikosR) [reply remark-lint-nodejs-yaml-comments.js:128]: No rule that's currently used in the repo. I use xo and it flagged this for me.

File: test/test.js [hunk 6]
```diff
@@ -1,13 +1,12 @@
-import assert from "assert";
+import { strict as assert } from "assert";
```
Comment (aduh95) [inline test/test.js:1]: I'm -0 on this change, I find `assert.strictEqual` clearer on its intent.

**Pull request closed without merging**